### PR TITLE
Features/v5/placeholders

### PR DESCRIPTION
**Pull request by Sauilitired**
Make placeholders less annoying and error prone. This will also allow us to add support for other placeholder APIs.

### CODE REVIEW

**Comment by Sauilitired** [comment]
This needs testing.

File: Core/src/main/java/com/plotsquared/core/util/placeholders/PlaceholderRegistry.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+                return "0";
+            }
+            return String.valueOf(plot.getDenied().size());
+        });
+        this.createPlaceholder("has_build_rights", (player, plot) ->
+            plot.isAdded(player.getUUID()) ? "true" : "false");
+        this.createPlaceholder("currentplot_x", (player, plot) -> Integer.toString(plot.getId().getX()));
+        this.createPlaceholder("currentplot_y", (player, plot) -> Integer.toString(plot.getId().getY()));
+        this.createPlaceholder("currentplot_xy", (player, plot) -> plot.getId().toString());
+        this.createPlaceholder("currentplot_rating", (player, plot) -> Double.toString(plot.getAverageRating()));
```
Comment (N0tMyFaultOG) [inline Core/src/main/java/com/plotsquared/core/util/placeholders/PlaceholderRegistry.java:119]: Does the rating placeholder work when you are using the like system?

Comment (Sauilitired) [reply Core/src/main/java/com/plotsquared/core/util/placeholders/PlaceholderRegistry.java:119]: yes, uses the same system internally

**Comment by Sauilitired** [comment]
just to be clear, this is currently broken and we need to delay hooking into mvdw until the server has loaded (ServerLoadEvent or w/e)

**Pull request merged**